packages/harmony/src/components/expandable-nav-item/ExpandableNavItem.tsx

```diff
@@ -0,0 +1,114 @@
+import { useMemo, useState } from 'react'
+
+import { useTheme, CSSObject } from '@emotion/react'
+
+import { HarmonyTheme } from '../../foundations/theme'
+import { IconCaretDown, IconCaretRight } from '../../icons'
+import { Flex } from '../layout/Flex'
+import { Text } from '../text'
+
+import type { ExpandableNavItemProps } from './types'
+
+const getStyles = (
+  theme: HarmonyTheme,
+  isOpen: boolean,
+  isHovered: boolean
+): CSSObject => {
+  const baseStyles: CSSObject = {
+    transition: `background-color ${theme.motion.hover}`,
+    cursor: 'pointer',
+    border: `1px solid ${isOpen ? theme.color.border.default : 'transparent'}`
+  }
+
+  const hoverStyles: CSSObject = {
+    backgroundColor: theme.color.background.surface2
+  }
+
+  return {
+    ...baseStyles,
+    ...(isHovered && hoverStyles)
+  }
+}
+
+export const ExpandableNavItem = ({
+  label,
+  leftIcon: LeftIcon,
+  rightIcon,
+  defaultIsOpen = false,
+  nestedItems,
+  shouldPersistRightIcon = false,
+  ...props
+}: ExpandableNavItemProps) => {
+  const [isOpen, setIsOpen] = useState(defaultIsOpen)
+  const [isHovered, setIsHovered] = useState(false)
+  const theme = useTheme()
+
+  const handleMouseEnter = () => setIsHovered(true)
+  const handleMouseLeave = () => setIsHovered(false)
+  const handleClick = () => setIsOpen(!isOpen)
+
+  const styles = useMemo(
+    () => getStyles(theme, isOpen, isHovered),
+    [theme, isOpen, isHovered]
+  )
+
+  const IconComponent = isHovered
+    ? isOpen
+      ? IconCaretDown
+      : IconCaretRight
+    : LeftIcon
+
+  return (
+    <Flex direction='column' {...props}>
```

Comment:
Lets also consider the a11y of this element. Do a bit of research, but we are probably looking at role='navigation' for the whole nav, and then these sub elements, potentially ul and li etc.